code/run_templates/etl_with_local_file/etl.py

```diff
@@ -0,0 +1,197 @@
+                dataset = dataset_api_instance.find_dataset_by_id(
+                    organization_id=os.environ.get("CSM_ORGANIZATION_ID"),
+                    dataset_id=value)
+                param_values = dataset['connector']['parameters_values']
+
+                # Check if file is AZ storage based and is a workspace file
+                _base_file_name = param_values.get('AZURE_STORAGE_CONTAINER_BLOB_PREFIX', False)
+                if not _base_file_name or "%WORKSPACE_FILE%" not in _base_file_name:
+                    LOGGER.warning(f"Parameter '{param_id}' is not a downloadable dataset - Skipping it")
+                    continue
+                LOGGER.info(f"Parameter '{param_id}' is a downloadable dataset")
+                _file_name = _base_file_name.replace('%WORKSPACE_FILE%/', '')
+                tmp_dataset_dir = os.environ.get("CSM_PARAMETERS_ABSOLUTE_PATH") + "/" + param_id
+                os.makedirs(tmp_dataset_dir, exist_ok=True)
+                # Lookup in previously loaded list of files to find all files corresponding to the dataset
+                existing_files = list(_f.to_dict().get('file_name')
+                                      for _f in all_api_files
+                                      if _f.to_dict().get('file_name', '').startswith(_file_name))
+                files_len = len(existing_files)
+                files_txt, those_txt = ["file", "files"][files_len > 1], ["it", "those"][files_len > 1]
+                LOGGER.info(f"  - {len(existing_files)} {files_txt} found in the dataset, downloading {those_txt}")
+                # Download and write locally all files corresponding
+                for _existing_file_name in existing_files:
+                    dl_file = workspace_api_instance.download_workspace_file(
+                        file_name=_existing_file_name,
+                        organization_id=os.environ.get("CSM_ORGANIZATION_ID"),
+                        workspace_id=os.environ.get("CSM_WORKSPACE_ID"))
+                    target_file = os.path.join(tmp_dataset_dir, _existing_file_name.split('/')[-1])
+                    with open(target_file, "wb") as tmp_file:
+                        tmp_file.write(dl_file.read())
+                value = tmp_dataset_dir
+            parameters[param_id] = (value, var_type)
+    # Now all parameters and Datasets are available locally
+    # %DATASETID% type files have value replaced with folder containing them
+
+    LOGGER.info("All parameters are loaded")
+
+    bars = list()
+
+    bar = {
+        "type": "Bar",
+        "name": "MyBar",
+        "params": f"""NbWaiters: {int(parameters["num_waiters"][0])},""" +
+                  f"""RestockQty: {int(parameters["restock_quantity"][0])},""" +
+                  f"""Stock: {int(parameters["stock"][0])}""",
+    }
+    bars.append(bar)
+
+    base_path = parameters["bar_instance"][0]
+    with ZipFile(base_path + "/brewery_instance.zip") as zip:
```

Comment:
Maybe replace fixed name by a glob in case user did not exactly named the file `brewery_instance.zip`

Reply:
but what if there's multiple file, even this a *.zip it's possible to have mutliple .zip file

Reply:
Avec le front il n'y aura qu'un seul zip à la fois donc tu peux récupérer le [0] du glob

Reply:
Done